OmsAgent/omsagent.py

```diff
@@ -251,6 +277,9 @@ def install():
     workspaceId = public_settings.get('workspaceId')
     check_workspace_id(workspaceId)
 
+    # Take the backup of the state for given workspace. 
+    restore_state(workspaceId)
```

Comment:
Shouldn't we restore in remove() too ? As the guest agent currently calls remove followed by install to update/upgrade the extension.